### PR TITLE
Fix isWellFormedDeep algorithm for timeline events

### PR DESCRIPTION
**Pull request by kenzieschmoll**
fix an issue discovered while testing 0.1.14-dev.1

### CODE REVIEW

File: packages/devtools_app/lib/src/timeline/timeline_model.dart [hunk 1]
```diff
@@ -668,13 +669,7 @@ abstract class TimelineEvent extends TreeNode<TimelineEvent> {
...
 
   bool _isWellFormedDeep(TimelineEvent event) {
-    if (!event.isWellFormed) {
-      return false;
-    }
-    for (var child in event.children) {
-      return _isWellFormedDeep(child);
-    }
-    return true;
+    return !containsChildWithCondition((e) => !e.isWellFormed);
```
Comment (jacob314) [inline packages/devtools_app/lib/src/timeline/timeline_model.dart]: aren't you now missing verifying that the root node is well formed?

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/timeline/timeline_model.dart]: The first node checked is the root

Comment (jacob314) [reply packages/devtools_app/lib/src/timeline/timeline_model.dart]: "containsChildWithCondition" is a bit of a misleading name.
How about, deepWithCondition or subtreeHasNodeWithCondition.
constainsChildWithCondition doesn't make it clear it is deep or that it includes the root node as well as the children.

**Comment by jacob314** [comment]
lgtm

File: packages/devtools_app/lib/src/timeline/timeline_model.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/timeline/timeline_model.dart]: renamed method to `subtreeHasNodeWithCondition`

**Pull request merged**